lib/mongo/client_encryption.rb

```diff
@@ -58,6 +58,27 @@ def close
       true
     end
 
+    # Generates a data key used for encryption/decryption and stores
+    # that key in the KMS collection. The generated key is encrypted with
+    # the KMS master key.
+    #
+    # @return [ BSON::Binary ] UUID representing the data key _id
+    def create_data_key
+      result = nil
+
+      Crypt::DataKeyContext.with_context(@crypt_handle.ref) do |context|
+        result = context.run_state_machine
+      end
+
+      data_key = Hash.from_bson(
+        BSON::ByteBuffer.new.put_bytes(result)
+      )
+
+      insert_result = @client.use(@key_vault_db)[@key_vault_coll].insert_one(data_key)
```

Comment:
I expect "data key" to be a literal value, like a string. I created https://jira.mongodb.org/browse/SPEC-1490 since I was unable to locate the type of data key in the spec.

Reply:
Thanks! I'll write a comment to make that clearer.

Reply:
Actually, I'll rename the variable to make it clear that it's a document.